### ISSUE TITLE
Quick LPCNet update

### ISSUE THREAD

**Issue by jmvalin** [xiph/LPCNet#181]
Just a few updates around LPCNet. The repo has been transferred from the mozilla organization to xiph. Everything should just keep working (let us know if it doesn't) as usual. 

Recently there have been many changes to the code. One set of changes is meant to make the synthesis even more efficient, with an equivalent reduction of around 3.5x in CPU for the same quality. The efficiency improvements are documented in a paper that was recently accepted for ICASSP 2022:
* J.-M. Valin, U. Isik, P. Smaragdis, A. Krishnaswamy, [Neural Speech Synthesis on a Shoestring: Improving the Efficiency of LPCNet](https://arxiv.org/pdf/2202.11169.pdf), *Accepted for ICASSP 2022*.

A second set of changes makes the LPC part fully-differentiable and should make it possible to train LPCNet end-to-end with any set of input features. The changes (which are optional at training time) are documented in the following pre-print:
* K. Subramani, J.-M. Valin, U. Isik, P. Smaragdis, A. Krishnaswamy, [End-to-end LPCNet: A Neural Vocoder With Fully-Differentiable LPC Estimation](https://arxiv.org/pdf/2202.11301.pdf), arxiv:[2106.04129](https://arxiv.org/abs/2202.11301), 2022.

**Comment by stayforapple** [comment]
Could you please develop an open-source ASIC (Application Specific Integrated Circuit) just for LPCNet ? Such that the synthesis can run in a cheap IC, which is good for the popularity of LPCNet.

**Comment by stayforapple** [comment]
And, how is the quality of the 1.6kbps vocoder with your new LPCNet ？

**Comment by shaun95** [comment]
Thanks for the heads up, for some reason I am not able to locate the branches cited in the two papers, lpcnet end2end and lpcnet efficiency

**Comment by yanggeng1995** [comment]
It's really a great work! I have a few questions about this paper that I hope you can answer.
"With hierarchical sampling, we cannot directly manipulate individual sample probabilities. Instead, each branching decision is biased to render very low probability events impossible. This can be achieved by seeding our pre-computed table with r, We find that ξ = 0.025 provides natural synthesis while reducing noise and removing the occasional glitch in the synthesis"

How to understand the concept of pre-computed table，I can't figure out the sampling process.

**Comment by acsuwut** [comment]
Hey @jmvalin -

Are the changes you're referring to in the start of this thread in the repository now?

Also, is there any formal roadmap for plans you have for this repository?

Thanks!

**Comment by jmvalin** [comment]
Yes, all of the changes I mentioned are currently in master. There is no formal roadmap at this point.

**Comment by acsuwut** [comment]
> Yes, all of the changes I mentioned are currently in master. There is no formal roadmap at this point.

Awesome, thank you for the quick response.

**Comment by SolomidHero** [comment]
Hi @jmvalin !
Is it correct, that by default any new trained model uses 1)efficiency improvement and 2)differential LPC computation?
Also what about models listed there https://media.xiph.org/lpcnet/data/ ?
Do they use above improvements if listed after April 2022?
